File: packages/typescript/src/services/text.ts
import { BaseService } from './base.js';

export interface SentimentResult {
  sentiment: 'positive' | 'negative' | 'neutral';
  score: number;
  confidence: number;
}

export interface SummaryResult {
  summary: string;
  original_length: number;
  summary_length: number;
  compression_ratio: number;
  key_points: string[];
}

export interface TranslationResult {
  translated_text: string;
  source_language: string;
  target_language: string;
  confidence: number;
}

export interface GrammarResult {
  corrected_text: string;
  corrections: Array<{ original: string; corrected: string; rule?: string }>;
}

export interface AutocompleteResult {
  completions: string[];
}

export interface VectorSearchContext {
  text: string;
  [key: string]: unknown;
}

export interface VectorSearchResult {
  results: Array<{ content: string; score: number; metadata?: Record<string, unknown> }>;
  total: number;
}

export class TextService extends BaseService {
  async sentiment(text: string): Promise<SentimentResult> {
    return this.request<SentimentResult>('POST', '/v1/text/sentiment', {
      body: { text },
    });
  }

  async summarize(
    text: string,
    options?: { length?: 'short' | 'medium' | 'long'; style?: 'bullet' | 'paragraph' },
  ): Promise<SummaryResult> {
    return this.request<SummaryResult>('POST', '/v1/text/summarize', {
      body: { text, ...options },
    });
  }

  async translate(
    text: string,
    targetLanguage: string,
    options?: { sourceLanguage?: string },
  ): Promise<TranslationResult> {
    return this.request<TranslationResult>('POST', '/v1/text/translate', {
      body: {
        text,
        target_language: targetLanguage,
        ...(options?.sourceLanguage ? { source_language: options.sourceLanguage } : {}),
      },
    });
  }

  async grammar(text: string): Promise<GrammarResult> {
    return this.request<GrammarResult>('POST', '/v1/text/grammar', {
      body: { text },
    });
  }

  async autocomplete(
    text: string,
    options?: { maxCompletions?: number },
  ): Promise<AutocompleteResult> {
    return this.request<AutocompleteResult>('POST', '/v1/text/autocomplete', {
      body: { text, ...options },
    });
  }

  async vectorSearch(
    query: string,
    context: VectorSearchContext[],
    options?: { topK?: number; threshold?: number },
  ): Promise<VectorSearchResult> {
    return this.request<VectorSearchResult>('POST', '/v1/text/vector-search', {
      body: { query, context, ...options },
    });
  }
}

File: packages/python/scalix_sdk/services/text.py
"""Text utilities — sentiment, summarization, translation, grammar, autocomplete, vector search."""

from __future__ import annotations

from typing import Any

from scalix_sdk.services.base import BaseService


class TextService(BaseService):
    async def sentiment(self, text: str) -> dict[str, Any]:
        return await self._request("POST", "/v1/text/sentiment", json={"text": text})

    async def summarize(
        self,
        text: str,
        *,
        length: str | None = None,
        style: str | None = None,
    ) -> dict[str, Any]:
        body: dict[str, Any] = {"text": text}
        if length:
            body["length"] = length
        if style:
            body["style"] = style
        return await self._request("POST", "/v1/text/summarize", json=body)

    async def translate(
        self,
        text: str,
        target_language: str,
        *,
        source_language: str | None = None,
    ) -> dict[str, Any]:
        body: dict[str, Any] = {"text": text, "target_language": target_language}
        if source_language:
            body["source_language"] = source_language
        return await self._request("POST", "/v1/text/translate", json=body)

    async def grammar(self, text: str) -> dict[str, Any]:
        return await self._request("POST", "/v1/text/grammar", json={"text": text})

    async def autocomplete(
        self,
        text: str,
        *,
        max_completions: int | None = None,
    ) -> dict[str, Any]:
        body: dict[str, Any] = {"text": text}
        if max_completions is not None:
            body["max_completions"] = max_completions
        return await self._request("POST", "/v1/text/autocomplete", json=body)

    async def vector_search(
        self,
        query: str,
        context: list[dict[str, Any]],
        *,
        top_k: int | None = None,
        threshold: float | None = None,
    ) -> dict[str, Any]:
        body: dict[str, Any] = {"query": query, "context": context}
        if top_k is not None:
            body["top_k"] = top_k
        if threshold is not None:
            body["threshold"] = threshold
        return await self._request("POST", "/v1/text/vector-search", json=body)
